### ISSUE TITLE
strokeBounds is incorrect for certain paths

### ISSUE THREAD

**Issue by adroitwhiz** [paperjs/paper.js#1824]
# Description/Steps to reproduce

In some cases, paths' `strokeBounds` will be incorrectly sized too small.
This can easily be seen with e.g. a rotated ellipse, as in the sketch below.

1. Create a `paper.Path.Ellipse`
2. Rotate it
3. Get its stroke bounds
4. Notice that the bounds do not cover the actual stroke (and vary depending on the rotation)

# Link to reproduction test-case

[Sketch](http://sketch.paperjs.org/#V/0.12.7/S/XZBPC8IwDMW/SullE8aYiiAbXhTvogcPboeuC26spqONCIrf3WxD/NNDSn59ebz0IVFdQKby0ALpWkZS26rvtUVPAsRKINxEpzpw8U5RHW+NaToP4SNHwaezDVIqTtMkiQSXIhq5b+7AeMF0mRQ5PidZjjlC7MnZFo5NRTWbz2fZB26ssY5hUBql22B4cZYUQbhIxvkxVmmvWHlWvifXA/hSOND0k/1Q8xXvGSs8GwhHi960l/6lmv7ir1xXCDL+pNKBaofVvUxPxfMF)

# Expected result

An item's `strokeBounds` should cover the entire item-- otherwise, this can lead to viewboxes of exported SVGs being incorrectly shrunken and cutting off outlines.

# Additional information

This occurs across browsers/OSes

**Comment by lehni** [comment]
@adroitwhiz yes that's a bug indeed! Your fix looks good. I'll do some testing shortly, but it looks good to go! Out of curiosity: Github says `LLK/paper.js` is 91 commits ahead. Is there other work that we should bring over?

**Comment by adroitwhiz** [comment]
There's a lot of stuff related to text import/export, most of which (I think) is specific to the Scratch paint editor.

There's a couple other things that may be worth merging:
- https://github.com/LLK/paper.js/pull/5 - add an option to allow hit tests to hit paths with no fill (plus [a follow-up](https://github.com/LLK/paper.js/pull/9) and [to-be-merged addition that does the same thing for paths with no stroke](https://github.com/LLK/paper.js/pull/39))
- https://github.com/LLK/paper.js/pull/28 - fix style merging in SVG export; I'm working on a more complete version of this that I plan to submit here as well

@fsih Any other LLK-specific paper.js fixes that could be upstreamed here?

**Comment by lehni** [comment]
@adroitwhiz apologies for not taking care of this yet. Unfortunately I have had very little time to take care of Paper.js in the past year due to a difficult personal situation. May I ask why you closed the issue?

**Comment by adroitwhiz** [comment]
PR #1825 fixed this specific issue. I can leave it open to discuss the other LLK-specific changes if you want.

**Comment by lehni** [comment]
Ah good. Perhaps we should close this and create follow up issues for the other bits?

**Comment by adroitwhiz** [comment]
Sounds good.
